Without the following patches illumos-gate build fail with errors like this:
In file included from /usr/include/dbus-1.0/dbus/dbus.h:30,                                                                          
                 from ../common/libhal.c:35:                                                                                         
/usr/include/dbus-1.0/dbus/dbus-address.h:74:14: error: expected ';' before 'void'                                                   
   74 | static inline void                                                                                                                                                                    
      |              ^~~~~                                                                                                           
      |              ;                                                                                                               

--- dbus-1.16.2/dbus/dbus-address.h.orig
+++ dbus-1.16.2/dbus/dbus-address.h
@@ -73,12 +73,9 @@
  * pointer_to_entries must not be #NULL, but *pointer_to_entries
  * may be #NULL.
  */
-static inline void
-dbus_clear_address_entries (DBusAddressEntry ***pointer_to_entries)
-{
-  _dbus_clear_pointer_impl (DBusAddressEntry *, pointer_to_entries,
-                            dbus_address_entries_free);
-}
+#define dbus_clear_address_entries(pointer) \
+	_dbus_clear_pointer_impl(DBusAddressEntry *, \
+	(pointer), dbus_address_entries_free)
 
 /** @} */
 
--- dbus-1.16.2/dbus/dbus-connection.h.orig
+++ dbus-1.16.2/dbus/dbus-connection.h
@@ -466,12 +466,9 @@
  * pointer_to_connection must not be #NULL, but *pointer_to_connection
  * may be #NULL.
  */
-static inline void
-dbus_clear_connection (DBusConnection **pointer_to_connection)
-{
-  _dbus_clear_pointer_impl (DBusConnection, pointer_to_connection,
-                            dbus_connection_unref);
-}
+#define dbus_clear_connection(pointer) \
+	_dbus_clear_pointer_impl(DBusConnection, \
+	(pointer), dbus_connection_unref)
 
 /** @} */
 
--- dbus-1.16.2/dbus/dbus-message.h.orig
+++ dbus-1.16.2/dbus/dbus-message.h
@@ -387,12 +387,8 @@
  * pointer_to_message must not be #NULL, but *pointer_to_message
  * may be #NULL.
  */
-static inline void
-dbus_clear_message (DBusMessage **pointer_to_message)
-{
-  _dbus_clear_pointer_impl (DBusMessage, pointer_to_message,
-                            dbus_message_unref);
-}
+#define dbus_clear_message(pointer) \
+	_dbus_clear_pointer_impl(DBusMessage, (pointer), dbus_message_unref)
 
 /** @} */
 
--- dbus-1.16.2/dbus/dbus-pending-call.h.orig
+++ dbus-1.16.2/dbus/dbus-pending-call.h
@@ -86,12 +86,9 @@
  * pointer_to_pending_call must not be #NULL, but *pointer_to_pending_call
  * may be #NULL.
  */
-static inline void
-dbus_clear_pending_call (DBusPendingCall **pointer_to_pending_call)
-{
-  _dbus_clear_pointer_impl (DBusPendingCall, pointer_to_pending_call,
-                            dbus_pending_call_unref);
-}
+#define dbus_clear_pending_call(pointer) \
+	_dbus_clear_pointer_impl(DBusPendingCall, (pointer), \
+	dbus_pending_call_unref)
 
 /** @} */
 
--- dbus-1.16.2/dbus/dbus-server.h.orig
+++ dbus-1.16.2/dbus/dbus-server.h
@@ -114,11 +114,8 @@
  * pointer_to_server must not be #NULL, but *pointer_to_server
  * may be #NULL.
  */
-static inline void
-dbus_clear_server (DBusServer **pointer_to_server)
-{
-  _dbus_clear_pointer_impl (DBusServer, pointer_to_server, dbus_server_unref);
-}
+#define dbus_clear_server(pointer) \
+	_dbus_clear_pointer_impl(DBusServer, (pointer), dbus_server_unref)
 
 /** @} */
 
